### PR TITLE
[Lua, SQL] Fix bugs in confrontation, add IDs to valkurm

### PR DESCRIPTION
**Pull request by Flibe-XI**
**_I affirm:_**
- [x] I understand that if I do not agree to the following points by completing the checkboxes my PR will be ignored.
- [x] I understand I should leave resolving conversations to the LandSandBoat team so that reviewers won't miss what was said.
- [x] I have read and understood the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/LandSandBoat/server/blob/base/CODE_OF_CONDUCT.md).
- [x] I have _**tested my code and the things my code has changed**_ since the last commit in the PR and will test after any later commits.

## What does this pull request do?

Improves a number of things required for confrontations to work properly
1) Add function annotations
2) Addresses some edge case bugs that cause confrontation to break down when players re-enter the zone without confrontation effect
3) Add time limit functionality to confrontations
4) Prepare player list for appropriate enmity list management on pop (To be added to npcUtil.lua)
5) Update confrontation effect flag to include xi.effectFlag.ON_ZONE
6) Add some mob IDs to valkurm IDs.lua for pirates chart quest
7) Add some text IDs to outer horutoto ruins IDs.lua

## Steps to test these changes
Setup some dummy confrontation somewhere, such as qm4 in valkurm dunes (!pos -168 4 -131 103, id = 17199740)
Test out numerous edge cases for wins and losses.  
Player can still force win with multiple players because hate list doesn't autopopulate all alliance members.  So spawning player can pop mobs, then zone.  Mobs will then despawn, and because valid player count > 0, win condition triggers.  This can be fixed by adjusting the logic in npcUtil.lua to include the enmity player list logic.  To be handled in separate PR.

coauthor: TracentEden <92269743+TracentEden@users.noreply.github.com>
github.com/AirSkyBoat/AirSkyBoat 0c8cb7e7fa7a28106acaf06fb85eb7fd878eeaa6
github.com/AirSkyBoat/AirSkyBoat 3881abe4437a6f428cfbb6ada9efda24e7d6ed61

### CODE REVIEW

File: scripts/zones/Valkurm_Dunes/IDs.lua [hunk 1]
```diff
@@ -51,14 +51,19 @@ zones[xi.zone.VALKURM_DUNES] =
...
-        GOLDEN_BAT      = GetFirstID('Golden_Bat'),
-        MARCHELUTE      = GetFirstID('Marchelute'),
-        DOMAN           = GetFirstID('Doman'),
-        ONRYO           = GetFirstID('Onryo'),
+        VALKURM_EMPEROR     = GetFirstID('Valkurm_Emperor'),
+        GOLDEN_BAT          = GetFirstID('Golden_Bat'),
+        MARCHELUTE          = GetFirstID('Marchelute'),
+        DOMAN               = GetFirstID('Doman'),
+        ONRYO               = GetFirstID('Onryo'),
+        HOUU_THE_SHOALWADER = GetFirstID('Houu_the_Shoalwader'),
```
Comment (KnowOne134) [inline scripts/zones/Valkurm_Dunes/IDs.lua]: can we put these in alpha order

Comment (Flibe-XI) [reply scripts/zones/Valkurm_Dunes/IDs.lua]: I can do that - Is that the format going forward?  I haven't seen any "IDs.lua" file follow that format yet, but it makes sense to me

Comment (Xaver-DaRed) [reply scripts/zones/Valkurm_Dunes/IDs.lua]: If we use a raw id, like with text ids, we follow the id order.
With this ones we are moving away from raw ids, so we have to follow another order. Alphabetical order makes sense

**Review by zach2good — APPROVED** [review]
I've purged all information about this system from my mind since I originally wrote it, so I'm going to have to assume this all does as you say it does 😆

**Comment by zach2good** [comment]
To get proper GH attribution, you have to add this line to the commit message:

```
Co-Authored-By: TracentEden <92269743+TracentEden@users.noreply.github.com>
```

Otherwise it won't pick it up

**Comment by zach2good** [comment]
Can you make sure to test other things that use confrontation? Garrison, the AMK Cardian Fight, etc.?

**Comment by HolmesIV** [comment]
> To get proper GH attribution, you have to add this line to the commit message:
> 
> ```
> Co-Authored-By: TracentEden <92269743+TracentEden@users.noreply.github.com>
> ```
> 
> Otherwise it won't pick it up

Thanks for that, will fix.

**Review by Xaver-DaRed — APPROVED** [review]

**Comment by Flibe-XI** [comment]
> To get proper GH attribution, you have to add this line to the commit message:
> 
> ```
> Co-Authored-By: TracentEden <92269743+TracentEden@users.noreply.github.com>
> ```
> 
> Otherwise it won't pick it up

Got it thanks.  Will fix tonight.
